.github/workflows/test.yml

```diff
@@ -79,3 +83,65 @@ jobs:
       env:
         LOCKFILE: gun1
       run: mix test test/tesla/adapter/gun_test.exs --trace
+
+  dialyzer:
+    runs-on: ubuntu-latest
+    name: Dialyzer
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      matrix:
+        include:
+          - otp: '25.3.2.12'
+            elixir: '1.15.8'
+            experimental: false
+          - otp: '26.2.5'
+            elixir: '1.16.3'
+            experimental: false
+          - otp: '27.0'
+            elixir: '1.17.1'
+            experimental: false
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Elixir
+      uses: erlef/setup-beam@v1
+      with:
+        elixir-version: ${{ matrix.elixir }}
+        otp-version: ${{ matrix.otp }}
+        version-type: strict
+    - name: Restore dependencies cache
+      uses: actions/cache@v4
+      id: cache
+      with:
+        path: |
+          deps
+          _build
+          dialyzer
+        key: ${{matrix.otp}}-${{ runner.os }}-dialyzer-${{ hashFiles('**/mix.lock') }}
+        restore-keys: |
+          ${{matrix.otp}}-${{ runner.os }}-dialyzer-
+    - name: Install Dependencies
+      if: steps.cache.outputs.cache-hit != 'true'
+      env:
+        MIX_ENV: test
+      run: |
+        mix local.rebar --force
+        mix local.hex --force
+        mix deps.get
+    # Doesn't cache PLTs based on mix.lock hash, as Dialyzer can incrementally update even old ones
+    # Cache key based on Elixir & Erlang version.
+    - name: Restore PLT cache
+      uses: actions/cache@v2
+      id: plt_cache
+      with:
+        key: |
+          ${{matrix.otp}}-${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-plt
+        restore-keys: |
+          ${{matrix.otp}}-${{ runner.os }}-${{ steps.beam.outputs.elixir-version }}-${{ steps.beam.outputs.otp-version }}-plt
```

Comment:
🤔  I "borrowed" this from another project -- not sure why we were using both `matrix.otp` and `steps.beam.outputs.otp-version`

Reply:
Always use `steps.beam.outputs.otp-version` across setups to avoid headaches. Probably it is OK to remove `matrix.otp` here

Reply:
okay. done!